docs/source/modules/event.rst

```diff
@@ -0,0 +1,99 @@
+Event Module
+============
+
+.. contents::
+    :depth: 3
+
+Description
+===========
+
+The Event module allows to create listener and assign action to event. The module is using :code:`MotechEvent` to provide
+event with specific data that can be sent by any MOTECH module (using :code:`EventRelay`) when an event is fired.
+The module is using ActiveMQ to support the event execution. It is possible to register your own callback
+function using :code:`EventCallbackService`.
+
+MotechEvent
+===========
+
+The Event module use :code:`MotechEvent` to store information about fired event. Instance of this class can be sent by
+any module to a specific handler to fire an event. Every module can also be a subscriber of an event and receiving events to fire.
+
+Main fields of :code:`MotechEvent`:
+
++-------------------+----------------------------------------------------------------+--------------------+
+|Field              |Description                                                     |Type                |
++===================+================================================================+====================+
+|subject            |Identifies event. The listeners can subscribe event using it.   |String              |
++-------------------+----------------------------------------------------------------+--------------------+
+|messageDestination |Destination of an event. Contains the id of the Motech listener |String              |
+|                   |that this event is meant for.                                   |                    |
++-------------------+----------------------------------------------------------------+--------------------+
+|callbackName       |Name of service, which is containing callback functions.        |String              |
++-------------------+----------------------------------------------------------------+--------------------+
+|parameters         |Parameters of event. Contains data used in event execution.     |Map<String, Object> |
++-------------------+----------------------------------------------------------------+--------------------+
+|metadata           |Stores any additional data, that is not meant to be the event's |Map<String, Object> |
+|                   |payload.                                                        |                    |
++-------------------+----------------------------------------------------------------+--------------------+
+
+This class is immutable.
+
+Listener
+========
+
+There are 2 ways to create an event listener:
+
+#. Use an :code:`@MotechListener` annotation to a handler function.
+
+    In :doc:`Core Architecture </architecture/core_architecture>` you can find an example of use of this annotation.
+
+#. Use an OSGi :code:`EventListenerRegistry` service exposed by the event module.
+
+Register custom callback
+========================
+
+ActiveMQ is sending events with its own retry. To register event callbacks you should implement an :code:`EventCallbackService`
```

Comment:
Expand the retry part. Mention that:
 - the default ActiveMQ retries will take place when the handler method fails (an exception is thrown)
 - the default retries can be disabled by custom event callback service (flag returned by the failureCallback method)